src/sortStudents.ts

```diff
@@ -1,16 +1,87 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(grades: number[]): number {
+  return grades.reduce((sum, num) => sum + num, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const newStudents: Student[] = students
+    .map((person: Student) => ({ ...person }));
+
+  const isAscending = order === 'asc';
+
+  switch (sortBy) {
+    case SortType.Name: {
+      return newStudents.sort(isAscending
+        ? (student1: Student, student2: Student): number => (
+          student1.name.localeCompare(student2.name)
+        )
+        : (student1: Student, student2: Student): number => (
+          student2.name.localeCompare(student1.name)
+        ));
+    }
+
+    case SortType.Surname: {
+      return newStudents.sort(isAscending
+        ? (student1: Student, student2: Student): number => (
+          student1.surname.localeCompare(student2.surname)
+        )
+        : (student1: Student, student2: Student): number => (
+          student2.surname.localeCompare(student1.surname)
+        ));
+    }
+
+    case SortType.Age: {
+      return newStudents.sort(isAscending
+        ? (student1: Student, student2: Student): number => (
+          student1.age - student2.age
+        )
+        : (student1: Student, student2: Student): number => (
+          student2.age - student1.age
+        ));
+    }
+
+    case SortType.Married: {
+      return newStudents.sort(isAscending
+        ? (student1: Student, student2: Student): number => (
+          Number(student1.married) - Number(student2.married)
+        )
+        : (student1: Student, student2: Student): number => (
+          Number(student2.married) - Number(student1.married)
+        ));
+    }
+
+    case SortType.AverageGrade: {
+      return newStudents.sort(isAscending
+        ? (student1: Student, student2: Student): number => (
+          getAverageGrade(student1.grades) - getAverageGrade(student2.grades)
+        )
+        : (student1: Student, student2: Student): number => (
+          getAverageGrade(student2.grades) - getAverageGrade(student1.grades)
+        ));
```

Comment:
does it seem like you are writing the same code twice? it does to me. you can leave the part that is the same and put the condition on the part that is not the same. my point is you don't need a separate callback. all you need is to change the order of student1 and student2.

```suggestion
    case SortType.AverageGrade: {
      return newStudents.sort((student1: Student, student2: Student): number => (
          isAscending
            ? getAverageGrade(student1.grades) - getAverageGrade(student2.grades)
            : getAverageGrade(student2.grades) - getAverageGrade(student1.grades)
      ));
```